content/components/hlk_fm22x.md

```diff
@@ -0,0 +1,332 @@
+```
+
+## `hlk_fm22x.enroll` Action
+
+Starts the face enrollment process with a name and direction.
+To successfully enroll a face, you need to successfully and consecutively scan the face from all directions.
+A failure in one direction will require enrolling the face again from the start.
+
+```yaml
+on_...:
+  then:
+    - hlk_fm22x.enroll:
+        name: "My name"
+        direction: 1
+    # Update the template text sensor for visual feedback
+    - text_sensor.template.publish:
+        id: face_state
+        state: "Look directly at the camera"
+```
+
+**Configuration options:**
+
+- **name** (**Required**, string, templatable): The name associated with the face. Up to 32 ASCII characters.
+- **direction** (**Required**, int, templatable): The direction to scan the face for. `1` for center, `2` for right, `4` for left, `8` for down, and `16` for up.
+
+## `hlk_fm22x.scan` Action
+
+Scans and tries to match to an enrolled face. Triggers one of the on_face_scan triggers.
+
+```yaml
+on_...:
+  then:
+    - hlk_fm22x.scan:
+```
+
+## `hlk_fm22x.delete` Action
+
+Removes the enrolled face from the slot number defined.
+
+```yaml
+on_...:
+  then:
+    - hlk_fm22x.delete:
+        face_id: 0
+    # Shorthand
+    - hlk_fm22x.delete: 0
+```
+
+**Configuration options:**
+
+- **face_id** (**Required**, int, templatable): The slot number of the enrolled face to delete.
+
+## `hlk_fm22x.delete_all` Action
+
+Removes all enrolled faces.
+
+```yaml
+on_...:
+  then:
+    - hlk_fm22x.delete_all:
+```
+
+## `hlk_fm22x.reset` Action
+
+Resets the module. Can be useful after a failed enrollment or scan if the module isn't responding correctly.
+If this command fails it will mark the module as failed.
+
+```yaml
+on_...:
+  then:
+    - hlk_fm22x.reset:
+```
+
+## All actions
+
+- **id** (*Optional*, ID): Manually specify the ID of the HLK-FM22x reader if you have multiple components.
+
+## Test setup
+
+With the following code you can quickly setup a node and use Home Assistant's action in the developer tools.
+E.g. for calling `hlk_fm22x.enroll` select the action `esphome.test_node_enroll` and in action data enter
+
+```json
+{ "name": "My name", "direction": 1 }
+```
+
+### Sample code
+
+```yaml
+uart:
+  rx_pin: GPIOXX
+  tx_pin: GPIOXX
+  baud_rate: 115200
+
+hlk_fm22x:
+  on_face_scan_invalid:
+    - homeassistant.event:
+        event: esphome.test_node_face_scan_invalid
+        data:
+          error: !lambda 'return error;'
+  on_face_scan_matched:
+    - homeassistant.event:
+        event: esphome.test_node_face_scan_matched
+        data:
+          face_id: !lambda 'return face_id;'
+          name: !lambda 'return name;'
+  on_face_scan_unmatched:
+    - homeassistant.event:
+        event: esphome.test_node_face_scan_unmatched
+  on_face_info:
+    - homeassistant.event:
+        event: esphome.test_node_face_info
+        data:
+          status: !lambda 'return status;'
+          left: !lambda 'return left;'
+          ...
```

Comment:
Looks like this is incomplete. Usually we expect users to be able to copy and paste it in and try it